### DISCUSSION TITLE
Data binding when "RelativeSource" is used in MVVM

### DISCUSSION THREAD

**Discussion by mwdpb** [canton7/Stylet#357] [Q&A] [answered]
I have a custom DataGrid with one column being ComboBox whose selected value is row-specific but its `ItemSource` should be globally bond.  The properties structure is:

VM level: 
```
PPTPlots {get; set}
```

Inside PPTPlots:
```
Items {get; set}
VisualTitles {get; set}
```

Inside VisualTitles:
```
Items {get; set}
```

and the DataGrid is written as below:

```
<DataGrid ItemsSource="{Binding PPTPlots.Items}">
    <DataGrid.Columns>
        <DataGridTemplateColumn>
            <DataGridTemplateColumn.CellTemplate>
                <DataTemplate>
                    <ComboBox ItemsSource="{Binding PPTPlots.VisualTitles.Items}"/>
                </DataTemplate>
            </DataGridTemplateColumn.CellTemplate>
        </DataGridTemplateColumn>
    </DataGrid.Columns>
</DataGrid>
```

The above code does not work as WPF is trying to bind `ItemSource` to `xxx.VisualTitle` (`xxx` is the type of PPTPlots.Items).  It basically uses the data row's context, instead of viewmodel's context.

So to remedy the problem, I changed the binding syntax to: 
```
<ComboBox ItemsSource="{Binding PPTPlots.VisualTitles.Items, RelativeSource={RelativeSource AncestorType=UserControl}}"/>
```

Now VS complained property not found in the view.  Since the property is defined in the viewmodel, it makes sense that it can't be found in the view.  So my question is in Stylet how can we make it looking for property in the view model (not in the view) when the `RelativeSource` clause is used?

![image](https://user-images.githubusercontent.com/55845355/190478015-5d90c341-fafb-4a69-aa3a-e4496362bfce.png)

**Comment by canton7** [comment, marked as answer]
The normal wpf way is `{Binding DataContext.PPTPlots....}`

**Reply by mwdpb** [reply]
What syntax should I use?  I tried below, and obviously it doesn't work.  For command binding, Stylet has `s.View.ActionTarget`, I was thinking if there's similar syntax to explicitly specify the datacontext.

![image](https://user-images.githubusercontent.com/55845355/190520462-30c59da5-dbed-4ec0-89cb-55d57d8b568f.png)

**Reply by canton7** [reply]
This is just a normal WPF binding, nothing to do with Stylet.

```
<ComboBox ItemsSource="{Binding DataContext.PPTPlots.VisualTitles.Items, RelativeSource={RelativeSource AncestorType=UserControl}}"/>
```

**Reply by canton7** [reply]
My normal trick here is to name the View:

```
<UserControl x:Name="root"...
```

Then use in the binding:

```
<ComboBox ItemsSource="{Binding DataContext.SomeVmProperty, ElementName=root}" 
```

**Reply by mwdpb** [reply]
Gotcha and it works great!  Thank you so much and a trick learned!  :)

**Reply by canton7** [reply]
Great!
